drivers/lg/displays/ls5.cr

```diff
@@ -0,0 +1,291 @@
+  end
+
+  def no_signal_off(state : Bool = false)
+    val = state ? 1 : 0
+    do_send(Command::NoSignalOff, val, 'f', name: "disable_no_signal_off")
+  end
+
+  def auto_off(state : Bool = false)
+    val = state ? 1 : 0
+    do_send(Command::AutoOff, val, 'm', name: "disable_auto_off")
+  end
+
+  def wake_on_lan(state : Bool = true)
+    val = state ? 1 : 0
+    do_send(Command::WakeOnLan, val, 'f', name: "enable_wake_on_lan")
+  end
+
+  def wake(broadcast : String? = nil)
+    if mac = setting?(String, :mac_address)
+      # config is the database model representing this device
+      wake_device(mac, broadcast)
+      logger.debug {
+        info = "Wake on Lan for MAC #{mac}"
+        if b = broadcast
+          info += " directed to VLAN #{b}"
+        end
+        info
+      }
+    else
+      logger.debug { "No MAC address provided" }
+    end
+  end
+
+  DUPLICATES = ["Contrast", "NoSignalOff", "AutoOff", "PmMode"]
+  def received(data, task)
+    command = Command.from_value(data[0]).to_s
+
+    # Both the responses for contrast/no_signal_off will have data[0] == 'g'
+    # Same thing for auto_off/pm_mode with data[0] == 'n'
+    # We will try using the task name to actually distinguish between these pairs
+    if DUPLICATES.includes?(command) && (task_name = task.try &.name)
+      case task_name
+      when .includes?("contrast")
+        command = "Contrast"
+      when .includes?("no_signal_off")
+        command = "NoSignalOff"
+      when .includes?("auto_off")
+        command = "AutoOff"
+      when .includes?("pm_mode")
+        command = "PmMode"
+      end
+    end
+    logger.debug { "Received command #{command}" }
+    data = String.new(data)
+    logger.debug { "LG sent #{data}" }
+
+    resp = data.split(' ').last
+
+    resp_value = 0
+    if resp[0..1] == "OK" # Extract the response value
+      # Special case for PM Mode
+      if resp[2..3] == "0c"
+        resp_value = resp[4..-2].to_i(16)
+      else
+        resp_value = resp[2..-2].to_i(16)
+      end
+    else # Request failed. We don't want to retry
+      return task.try &.abort
+    end
+
+    case command
+    when "Power"
+      self[:hard_power] = resp_value == 1
+      self[:power] = false unless self[:hard_power].as_bool
+    when "Input"
+      self[:input] = Input.from_value(resp_value)
+    when "AspectRatio"
+      self[:aspect_ratio] = Ratio.from_value(resp_value)
+    when "ScreenMute"
+      self[:power] = resp_value == 0
+    when "VolumeMute"
+      self[:audio_mute] = resp_value == 0
+    when "Contrast", "Brightness", "Sharpness", "Volume", "PmMode"
+      self[command.underscore] = resp_value
+    when "WakeOnLan", "NoSignalOff", "AutoOff"
+      self[command.underscore] = resp_value == 1
+    when "LocalButtonLock"
+      self[:local_button_lock] = resp_value == 2
+    else
+      return task.try &.retry
+    end
+
+    task.try &.success
+  end
+
+  private def do_send(command : Command, data : Int, system : Char = 'k', **options)
```

Comment:
What special about system `k`? Can this be given some meaning from either encoding this in a value, or at a minimum referencing in a comment?